### DISCUSSION TITLE
Error in installing and Configure the omnetpp 6.0.1

### DISCUSSION THREAD

**Discussion by adevelopera** [omnetpp/omnetpp#996] [Q&A] [answered]
I have trouble with installing omnetpp 6.0.1 
this is my issue:
__________________________________________________________________________________
configure: Environment variables (PATH and PYTHONPATH) are correctly set.
configure: Reading configure.user for your custom settings.
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for clang... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for clang++... clang++
checking whether we are using the GNU C++ compiler... yes
checking whether clang++ accepts -g... yes
checking for clang++... clang++
checking for c++14 support... yes
checking for ranlib... ranlib
checking whether LLD linker is available... no
checking whether clang++ supports -fno-omit-frame-pointer... no
checking whether clang++ supports -glldb... no
checking whether clang++ supports -ggdb3... no
checking whether clang++ supports -fstandalone-debug... no
checking whether clang++ supports -fno-limit-debug-info... no
checking whether clang++ supports -Wl,--no-as-needed... no
checking whether clang++ supports -Wl,--as-needed... no
checking for swapcontext... no
checking if shared libs need -fPIC... failure
configure: WARNING: COULD NOT FIGURE OUT HOW TO BUILD SHARED LIBRARY -- reverting to static libs
checking for dlopen with CFLAGS="" LIBS=""... no
checking if --export-dynamic linker option is supported/needed... test failed
checking for flags needed to link with static libs containing simple modules... no solution found
configure: WARNING: Avoid linking with static libraries containing simple modules
configure: WARNING: and other dynamically registered components. Those components
configure: WARNING: will not be found by the simulation program, because the linker
configure: WARNING: leaves them out from the executable.
checking whether linker supports -rpath... no
checking for bison... bison -y
checking for flex... flex
checking lex output file root... lex.yy
checking lex library... none needed
checking whether yytext is a pointer... no
checking for make... make
checking for perl... perl
checking for swig... not found
checking for python3... python3
checking if all necessary Python modules are available... yes
checking for math with CFLAGS="" LIBS=""... no
checking for math with CFLAGS="" LIBS="-lm"... no
configure: error: Standard C math library -lm not found
__________________________________________________________________________________
please help to solve.
this is very important to me.

**Comment by rhornig** [comment]
There is something weird going on here. with your system setup. While it complains about the math library missing, there is already a serious issue earlier that the configure script cannot detect how to build shared libraries. This is definitely not good. Take a look / post the `config.log` file which might give you further info. 

Be sure to install all required dependencies (see InastallGuide.pdf) before attempting to run configure

**Comment by adevelopera** [comment]
Thank you for your attention
I checked the config.log and part of the error is as follows:
___________________________________________________________________
1 warning generated.
/usr/bin/ld: cannot find -lstdc++: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure:4566: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| #include <math.h>
| int
| main ()
| {
| floor(0);
|   ;
|   return 0;
| }
configure:4570: result: no
configure:4587: checking for math with CFLAGS="" LIBS="-lm"
configure:4601: clang++ -o conftest    conftest.cpp -lm >&5
conftest.cpp:13:1: warning: ignoring return value of function declared with const attribute [-Wunused-value]
floor(0);
^~~~~ ~
1 warning generated.
/usr/bin/ld: cannot find -lstdc++: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure:4601: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| #include <math.h>
| int
| main ()
| {
| floor(0);
|   ;
|   return 0;
| }
configure:4605: result: no
configure:4616: error: Standard C math library -lm not found

**Reply by rhornig** [reply, marked as answer]
This means even the standard C++ library is not installed on your system. Depending on the distribution you have (you did not specify that) you should install all the packages required to build C++ programs. On debian based systems you I would install `build-essential`, `libstdc++6`. Generally, you should make sure that you have the necessary packages installed to build any C++ based programs.
